examples/custom_role_project/main.tf

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/******************************************
+  Provider configuration
+ *****************************************/
+provider "google" {
+  version = "~> 3.3"
+}
+
+provider "google-beta" {
+  version = "~> 3.3"
+}
+
+resource "random_id" "rand_custom_id" {
+  byte_length = 2
+}
+
+/******************************************
+  Module custom_role call
+ *****************************************/
+module "custom-role-project" {
+  source = "../../modules/custom_role_iam/"
+
+  target_level = "project"
+  target_id    = var.target_id
+  role_id      = "custom_role_project_${random_id.rand_custom_id.hex}"
```

Comment:
I don't think we need to make this random. Let's give it a descriptive name and make the role name meaningful.

Ex. we could give `iamDeleter` with permissions off `iam.roles.list` and `iam.roles.delete`.

Reply:
Done.